src/main/groovy/wooga/gradle/unity/APICompatibilityLevel.groovy

```diff
@@ -5,13 +5,56 @@ import org.gradle.internal.impldep.org.apache.commons.lang.NullArgumentException
 import java.nio.InvalidMarkException
 import java.security.InvalidKeyException
 
+/*
+.NET profile support
+
+Unity supports a number of .NET profiles. Each profile provides a different API surface for C# code
+which interacts with the .NET class libraries.
+
+The legacy scripting runtime supports two different profiles:
+NET 2.0 Subset and .NET 2.0. Both of these are closely aligned with the .NET 2.0 profile from Microsoft.
+The .NET 2.0 Subset profile is smaller than the .NET 4.x profile, and it allows access to the class library APIs that most Unity projects use.
+It is the ideal choice for size-constrained platforms, such as mobile, and it provides a set of portable APIs for multiplatform support.
+By default, most Unity projects should use the .NET Standard 2.0 profile.
+
+The stable scripting runtime supports two different profiles: .NET Standard 2.0 and .NET 4.x.
+The name of the .NET Standard 2.0 profile can be a bit misleading because it is not related to the .NET 2.0 and .NET 2.0 Subset profile
+from the legacy scripting runtime. Instead, Unity’s support for the .NET Standard 2.0 profile matches the profile of the same name published
+by the .NET Foundation. The .NET 4.x profile in Unity matches the .NET 4 series (.NET 4.5, .NET 4.6, .NET 4.7, and so on) of profiles from the .NET Framework.
+
+Only use the .NET 4.x profile for compatibility with external libraries, or when you require functionality that is not available in .NET Standard 2.0.
+
+Unity aims to support the vast majority of the APIs in the .NET Standard 2.0 profile on all platforms.
+While not all platforms fully support the .NET Standard, libraries which aim for cross-platform compatibility should target the .NET Standard 2.0 profile.
+The .NET 4.x profile includes a much larger API surface, including parts which may work on few or no platforms.
+ */
+
+enum BuildTargetGroup {
+    Standalone,
+    Android,
+    iPhone,
+}
```

Comment:
Is this list exaustive or just the values we use at wooga? There is already a `BuildTargetGroup` enum in this project. That being said. It is highly unportable to use because differerent Unity versions support different BuildTargetGroups. For now I think it is OK. But to have this as a general feature we would need to add support to set these groups directly. Means a Map setter in the task. Which would then mean that the `toSerializedMap` method would need to go as well.

Reply:
Just the list of targets we support; I think for now I would rename this enum to indicate its a subset.